File: app/src/main/java/com/boardgamegeek/ui/dialog/PlayStatsIncludeSettingsDialogFragment.java
package com.boardgamegeek.ui.dialog;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;

import com.boardgamegeek.R;
import com.boardgamegeek.util.PreferencesUtils;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;
import hugo.weaving.DebugLog;

public class PlayStatsIncludeSettingsDialogFragment extends DialogFragment {
	private ViewGroup root;
	private Unbinder unbinder;
	@BindView(R.id.incomplete) CheckBox includeIncompleteGamesView;
	@BindView(R.id.expansions) CheckBox includeExpansionsView;
	@BindView(R.id.accessories) CheckBox includeAccessoriesView;

	@DebugLog
	public PlayStatsIncludeSettingsDialogFragment() {
	}

	@DebugLog
	public static PlayStatsIncludeSettingsDialogFragment newInstance(@Nullable ViewGroup root) {
		final PlayStatsIncludeSettingsDialogFragment fragment = new PlayStatsIncludeSettingsDialogFragment();
		fragment.root = root;
		return fragment;
	}

	@DebugLog
	@Override
	@NonNull
	public Dialog onCreateDialog(Bundle savedInstanceState) {
		LayoutInflater layoutInflater = LayoutInflater.from(getActivity());
		View rootView = layoutInflater.inflate(R.layout.dialog_play_stats_settings_include, root, false);

		unbinder = ButterKnife.bind(this, rootView);
		bindUi();

		AlertDialog.Builder builder = new AlertDialog.Builder(getActivity(), R.style.Theme_bgglight_Dialog_Alert)
			.setView(rootView)
			.setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
				public void onClick(DialogInterface dialog, int id) {
					PreferencesUtils.putPlayStatsIncomplete(getActivity(), includeIncompleteGamesView.isChecked());
					PreferencesUtils.putPlayStatsExpansions(getActivity(), includeExpansionsView.isChecked());
					PreferencesUtils.putPlayStatsAccessories(getActivity(), includeAccessoriesView.isChecked());
				}
			});
		builder.setTitle(R.string.title_settings);
		return builder.create();
	}

	@Override
	public void onDestroyView() {
		super.onDestroyView();
		if (unbinder != null) unbinder.unbind();
	}

	private void bindUi() {
		includeIncompleteGamesView.setChecked(PreferencesUtils.logPlayStatsIncomplete(getActivity()));
		includeExpansionsView.setChecked(PreferencesUtils.logPlayStatsExpansions(getActivity()));
		includeAccessoriesView.setChecked(PreferencesUtils.logPlayStatsAccessories(getActivity()));
	}
}

File: app/src/main/java/com/boardgamegeek/ui/dialog/PlayStatsIncludeSettingsDialogFragment.kt
package com.boardgamegeek.ui.dialog

import android.annotation.SuppressLint
import android.app.Dialog
import android.os.Bundle
import androidx.fragment.app.DialogFragment
import androidx.appcompat.app.AlertDialog
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.boardgamegeek.R
import com.boardgamegeek.databinding.DialogPlayStatsSettingsIncludeBinding
import com.boardgamegeek.util.PreferencesUtils

class PlayStatsIncludeSettingsDialogFragment : DialogFragment() {
    private var _binding: DialogPlayStatsSettingsIncludeBinding? = null
    private val binding get() = _binding!!
    lateinit var layout: View

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        _binding = DialogPlayStatsSettingsIncludeBinding.inflate(LayoutInflater.from(context))
        layout = binding.root

        return AlertDialog.Builder(requireContext(), R.style.Theme_bgglight_Dialog_Alert)
                .setTitle(R.string.title_settings)
                .setView(layout)
                .setPositiveButton(R.string.ok) { _, _ ->
                    PreferencesUtils.putPlayStatsIncomplete(context, binding.includeIncompleteGamesView.isChecked)
                    PreferencesUtils.putPlayStatsExpansions(context, binding.includeExpansionsView.isChecked)
                    PreferencesUtils.putPlayStatsAccessories(context, binding.includeAccessoriesView.isChecked)
                }
                .create()
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        return layout
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding.includeIncompleteGamesView.isChecked = PreferencesUtils.logPlayStatsIncomplete(context)
        binding.includeExpansionsView.isChecked = PreferencesUtils.logPlayStatsExpansions(context)
        binding.includeAccessoriesView.isChecked = PreferencesUtils.logPlayStatsAccessories(context)
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    companion object {
        fun newInstance(): PlayStatsIncludeSettingsDialogFragment {
            return PlayStatsIncludeSettingsDialogFragment()
        }
    }
}
